src/main/java/com/faforever/moderatorclient/api/FafApiCommunicationService.java

```diff
@@ -141,9 +144,24 @@ public MeResult login(String username, String password) {
         }
     }
 
+    public void forceRenameUserName(String userId, String newName) {
+        String path = String.format("/users/%s/forceChangeUsername", userId);
+        String url = UriComponentsBuilder.fromPath(path).queryParam("newUsername", newName).toUriString();
+        try {
+            restTemplate.exchange(url, HttpMethod.POST, null, Void.class, Map.of());
+        } catch (Throwable t) {
+            applicationEventPublisher.publishEvent(new FafApiFailModifyEvent(t, Void.class, url));
+            throw t;
+        }
+    }
+
     @SneakyThrows
     public <T extends ElideEntity> T post(ElideNavigatorOnCollection<T> navigator, T object) {
-        String url = navigator.build();
+        return post(navigator.build(), navigator.getDtoClass(), object);
+    }
+
+    @Nullable
+    private <T, E> E post(String url, Class<E> returnType, T object) throws DocumentSerializationException, InterruptedException {
```

Comment:
This is used nowhere  as far as I can see? why do we need that method?

Reply:
We can revert this but this was my try of refactoring existing code to use it because right now only elide routes could be called